Source/Core/DolphinWX/ISOProperties.cpp

```diff
@@ -1353,6 +1397,97 @@ void CISOProperties::ListSelectionChanged(wxCommandEvent& event)
 	}
 }
 
+void CISOProperties::CheckboxSelectionChanged(wxCommandEvent& event)
+{
+	RmObjList_Save();
+	RmObjList_Load();
+}
+
+void CISOProperties::RmObjList_Load()
+{
+	rmObjCodes.clear();
+	RmObjs->Clear();
+
+	RmObjEngine::LoadRmObjSection("RmObjCodes", rmObjCodes, GameIniDefault, GameIniLocal);
+
+	u32 index = 0;
+	for (RmObjEngine::RmObj& p : rmObjCodes)
+	{
+		RmObjs->Append(StrToWxStr(p.name));
+		RmObjs->Check(index, p.active);
+		if (!p.user_defined)
+			DefaultRmObjs.insert(p.name);
+		++index;
+	}
+
+	RmObjEngine::ApplyRmObjs(rmObjCodes);
+
+}
+
+void CISOProperties::RmObjList_Save()
+{
+	std::vector<std::string> lines;
+	std::vector<std::string> enabledLines;
+	u32 index = 0;
+	for (RmObjEngine::RmObj& p : rmObjCodes)
+	{
+		if (RmObjs->IsChecked(index))
+			enabledLines.push_back("$" + p.name);
+
+		// Do not save default removed objects.
+		if (DefaultRmObjs.find(p.name) == DefaultRmObjs.end())
+		{
+			lines.push_back("$" + p.name);
+			for (const RmObjEngine::RmObjEntry& entry : p.entries)
+			{
+				std::string temp = StringFromFormat("%s:0x%08X%08X:0x%08X%08X", RmObjEngine::RmObjTypeStrings[entry.type], (entry.value_upper & 0xffffffff00000000) >> 32, (entry.value_upper & 0xffffffff), (entry.value_lower & 0xffffffff00000000) >> 32, (entry.value_lower & 0xffffffff));
+				lines.push_back(temp);
+			}
+		}
+		++index;
+	}
+	GameIniLocal.SetLines("RmObjCodes_Enabled", enabledLines);
+	GameIniLocal.SetLines("RmObjCodes", lines);
+}
+
+void CISOProperties::RmObjButtonClicked(wxCommandEvent& event)
+{
+	int selection = RmObjs->GetSelection();
+
+	switch (event.GetId())
+	{
+	case ID_EDITRMOBJ:
+	{
+		CRmObjAddEdit dlg(selection, this);
+		dlg.ShowModal();
+	}
+	break;
+	case ID_ADDRMOBJ:
+	{
+		CRmObjAddEdit dlg(-1, this, 1, _("Add Object to Remove"));
+		if (dlg.ShowModal() == wxID_OK)
+		{
+			RmObjs->Append(StrToWxStr(rmObjCodes.back().name));
+			RmObjs->Check((unsigned int)(rmObjCodes.size() - 1), rmObjCodes.back().active);
+		}
+	}
+	break;
+	case ID_REMOVERMOBJ:
+		rmObjCodes.erase(rmObjCodes.begin() + RmObjs->GetSelection());
+		RmObjs->Delete(RmObjs->GetSelection());
+		break;
+	}
+
+	RmObjList_Save();
+	RmObjs->Clear();
+	RmObjList_Load();
+
+	RmObjEngine::ApplyRmObjs(rmObjCodes);
+
+	EditRmObj->Enable(false);
```

Comment:
There are 10 other ->Enable(false)'s in ISOProperties (which is why I set it this way).  Is it bad form to change them all to ->Disable in this pull request or should I leave the others be?

Also, a lot of the .at()'s/push_back's etc, I also got from the "AR codes" files that I based my code on.  Should those be changed as well, or should those changes be done in a separate pull?